services/frontend-service/src/ui/ActionDetails.tsx

```diff
@@ -82,104 +118,181 @@ type ActionDetails = {
+            summary: 'Delete environment lock on ' + action.deleteEnvironmentLock.environment,
+            icon: <LockOpenRounded />,
+            environment: action.deleteEnvironmentLock.environment,
+            lockId: action.deleteEnvironmentLock.lockId,
+        };
+    if ('deleteApplicationLock' in action)
+        return {
+            type: ActionTypes.DeleteApplicationLock,
+            name: 'Delete App Lock',
+            dialogTitle: 'Are you sure you want to delete this application lock?',
+            summary:
+                'Unlock "' +
+                action.deleteApplicationLock.application +
+                '" on ' +
+                action.deleteApplicationLock.environment,
+            icon: <LockOpenRounded />,
+            environment: action.deleteApplicationLock.environment,
+            application: action.deleteApplicationLock.application,
+            lockId: action.deleteApplicationLock.lockId,
+        };
+    if ('prepareUndeploy' in action)
+        return {
+            type: ActionTypes.PrepareUndeploy,
+            name: 'Prepare Undeploy',
+            dialogTitle: 'Are you sure you want to start undeploy?',
+            description:
+                'The new version will go through the same cycle as any other versions' +
+                ' (e.g. development->staging->production). ' +
+                'The behavior is similar to any other version that is created normally.',
+            summary: 'Prepare undeploy version for Application ' + action.prepareUndeploy.application,
+            icon: <DeleteOutlineRounded />,
+            application: action.prepareUndeploy.application,
+        };
+    if ('undeploy' in action)
+        return {
+            type: ActionTypes.Undeploy,
+            name: 'Undeploy',
+            dialogTitle: 'Are you sure you want to undeploy this application?',
+            description: 'This application will be deleted permanently',
+            summary: 'Undeploy and delete Application ' + action.undeploy.application,
+            icon: <DeleteForeverRounded />,
+            application: action.undeploy.application,
+        };
+    else
+        return {
+            type: ActionTypes.UNKNOWN,
+            name: 'invalid',
+            dialogTitle: 'invalid',
+            summary: 'invalid',
+            icon: <Error />,
+        };
+};
+
+const randomLockId = () => 'ui-' + Math.random().toString(36).substring(7);
+
+export const transformToBatch = (act: CartAction, m: string): BatchAction => {
+    if ('createEnvironmentLock' in act)
+        return {
+            action: {
+                $case: 'createEnvironmentLock',
+                createEnvironmentLock: {
+                    environment: act.createEnvironmentLock.environment,
+                    lockId: randomLockId(),
+                    message: m,
+                },
+            },
+        };
+    if ('createApplicationLock' in act)
```

Comment:
I would use `else if`, like above, to more strongly emphasise that these blocks handle different cases for the same variable.